mitodl/mitxpro
# Added course enrollment button to course detail page

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes #67 

#### What's this PR do?
- Adds the edX demo course to the seed data
- Adds a simple course detail django template with a link to the edX course run and an enrollment button for course runs with `courseware_url_path` specified
- Enrolls a user in the given courserun when that button is clicked

#### How should this be manually tested?
- Run the `seed_data` management command
- Log in as a user with a `CoursewareUser` and `OpenEdxApiAuth` already associated
- Check that your user isn't enrolled in the edX demo course
- Navigate to the edX demo course detail page via the course catalog
- Click "Enroll" on the course run
- Make sure you're enrolled in that course on edX

#### Any background context you want to provide?
- I decided to add a course detail page instead of using the existing CMS course page. We may want to keep them separate (CMS page for marketing, other course detail page for enrolling/checking grades, etc.)
- The enrollment is implemented as a form submit. It seems likely that we'll move to a JS/React based approach for the course run UX for logged-in users, so it doesn't seem worth it to test the view that handles that form submit

#### Screenshots (if appropriate)
<img width="379" alt="ss 2019-03-28 at 15 03 23 " src="https://user-images.githubusercontent.com/14932219/55185887-adc17c80-516b-11e9-8f52-06c973641daa.png">
<img width="520" alt="ss 2019-03-28 at 15 03 27 " src="https://user-images.githubusercontent.com/14932219/55185888-adc17c80-516b-11e9-8a93-58da28d89e82.png">
